rllib/agents/sac/sac_tf_model.py

```diff
@@ -209,9 +209,15 @@ def get_policy_output(self, model_out: TensorType) -> TensorType:
         return self.action_model(model_out)
 
     def policy_variables(self):
-        """Return the list of variables for the policy net."""
+        """Return the list of variables for the policy net.
 
-        return list(self.action_model.variables)
+        This may include any variables present in the wrapping ModelV2 (e.g.
+        a Conv2D image preprocessor).
+        """
+
+        all_vars = set(self.trainable_variables()) - set(self.q_variables()) \
+                   - {self.log_alpha}
```

Comment:
Hmm does this mean Q variables don't include preprocessor vars? Would that prevent the loss from flowing from Q net to preprocessor?

Reply:
I want to avoid having two optimizers update the (currently shared) preprocessor.
I think the correct solution would be to have two different Conv2D stacks, one for the policy net and one for the Q-net, but that would require the user to write a custom model containing two separate Conv2D stacks. The RLlib default visual_net would not work here, then.

Reply:
What's the reason for that? Intuitively I would expect not propagating the loss from one stack to lead to worse behavior than optimizing it from two losses.